src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -309,21 +343,105 @@ private FormValidation checkAgentConnectivity(final String host, final String po
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, credentialsApiKey.getSecret());
+        } else {
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, Secret.fromString(targetApiKey));
+        }
+        if (validApiConnection) {
             return FormValidation.ok("Great! Your API key is valid.");
         } else {
             return FormValidation.error("Hmmm, your API key seems to be invalid.");
         }
     }
 
+    /**
+     * Populates the targetCredentialsApiKey field from the configuration screen with all of the valid credentials
+     *
+     * @param item - The context within which to list available credentials
+     * @param targetCredentialsApiKey - A String containing the API key as a credential
+     * @return a ListBoxModel object used to display all of the available credentials.
+     */
+    public ListBoxModel doFillTargetCredentialsApiKeyItems(
+        @AncestorInPath Item item,
+        @QueryParameter String targetCredentialsApiKey
+        ) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(targetCredentialsApiKey);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return result.includeCurrentValue(targetCredentialsApiKey);
+            }
+        }
+        return result.includeEmptyValue()
+            .includeMatchingAs(ACL.SYSTEM,
+                               Jenkins.get(),
+                               StringCredentials.class,
+                               Collections.emptyList(),
+                               CredentialsMatchers.instanceOf(StringCredentials.class))
+            .includeCurrentValue(targetCredentialsApiKey);
+    }
+
+
+    /**
+     * Tests the targetCredentialsApiKey field from the configuration screen, to check its' validity.
+     *
+     * @param item - The context within which to list available credentials.
+     * @param targetCredentialsApiKey - A String containing the API key as a credential
+     * @return a FormValidation object used to display a message to the user on the configuration
+     * screen.
+     */
+    @RequirePOST
+    public FormValidation doCheckCredentialsApiKey(
+        @AncestorInPath Item item,
+        @QueryParameter String targetCredentialsApiKey
+    ) {
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return FormValidation.ok();
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return FormValidation.ok(); 
+            }
+        }
```

Comment:
Could you add a comment explaining the logic?

Reply:
Added! The logic is based on the code snippet here: https://github.com/jenkinsci/credentials-plugin/blob/master/docs/consumer.adoc#providing-a-ui-form-element-to-let-a-user-select-credentials